pkg/transport/transport.go

```diff
@@ -94,7 +97,10 @@ func (transport *Transport) DialAddress(ctx context.Context, address string, opt
 	}
 
 	options := append([]grpc.DialOption{dialOpt, grpc.WithBlock()}, opts...)
-	conn, err = grpc.Dial(address, options...)
+	conn, err = grpc.DialContext(ctx, address, options...)
+	if err == context.Canceled {
```

Comment:
Can't we assume that if err != nil then conn == nil and remove this if statement?